lib/Sema/CodeSynthesisDistributedActor.cpp

```diff
@@ -346,17 +346,15 @@ deriveBodyDistributed_thunk(AbstractFunctionDecl *thunk, void *context) {
 
   // --- Recording invocation details
   // -- recordGenericSubstitution(s)
-  if (thunk->isGeneric() || nominal->isGeneric()) {
+  if (auto genEnv = thunk->getGenericEnvironment()) {
```

Comment:
You don't need a GenericEnvironment unless you want to produce archetypes. Maybe you want `if (auto genericSig = thunk->getGenericSignature())`

Reply:
Discussed some more offline, we ned the env -- but we can use it directly below in `mapTypeIntoContext`, I'll add that.

Thanks Slava!

Reply:
done here https://github.com/apple/swift/pull/71492/files#diff-c5ee3c123c92773a4f07b456c2c0051c7a9dfdc81c5c0234ba5da6389bd82a07R358